src/microbe_stage/MicrobeAI.cs

```diff
@@ -370,34 +376,43 @@ private void RunAndTumble(Random random)
         microbe.State = Microbe.MicrobeState.Normal;
 
         var usefulCompounds = microbe.TotalAbsorbedCompounds.Where(x => microbe.Compounds.IsUseful(x.Key));
-
-        // If this microbe lacks glucose, don't bother with ammonia and phosphorous
-        // This algorithm doesn't try to determine if iron and sulfuric acid is useful to this microbe
-        if (microbe.Compounds.GetCompoundAmount(glucose) < 0.5f)
-        {
-            usefulCompounds = usefulCompounds.Where(x => x.Key != ammonia && x.Key != phosphates);
-        }
+        var compoundsPriority = ComputeCompoundsPriorities(usefulCompounds.ToList());
 
         float compoundDifference = 0.0f;
-        foreach (var compound in usefulCompounds)
+        foreach (var compoundPriority in compoundsPriority.ToList())
         {
-            compoundDifference += compound.Value;
+            // Note this is about absorbed quantities (which is all microbe has access to), not real ones
+            float quantityDifference = 0.0f;
```

Comment:
I took a while to pick up on the difference between quantityDifference and compoundDifference, which is a good sign these need to be renamed more clearly.

Reply:
Agreed. I kept compoundDifference name legacy, but it seems to do more harm than good, because I don't see how to make quantityDifference clearer without renaming compondDifference aswell.

Reply:
No name is sacred. Code without mercy.